core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.spec.ts

```diff
@@ -323,13 +345,79 @@ describe('Topics and Skills Dashboard Page', function() {
       });
 
       ctrl.$onInit();
+      ctrl.displayedSkillSummaries = [];
       $rootScope.$apply();
     }));
 
     it('should change active tab to skills if topic summaries are null',
       function() {
-        expect(ctrl.activeTab).toEqual('untriagedSkills');
+        expect(ctrl.activeTab).toEqual('skills');
+      });
+
+    it('should fetch skills when filters are applied', function() {
+      expect(ctrl.activeTab).toEqual('skills');
+      var fetchSkillSpy = spyOn(ctrl, 'fetchSkills').and.callThrough();
+      ctrl.applyFilters();
+      expect(fetchSkillSpy).toHaveBeenCalled();
+    });
+
+    it('should paginate if skills are present in memory instead of fetching',
+      function() {
+        expect(ctrl.activeTab).toEqual('skills');
+        ctrl.more = true;
+        ctrl.fetchSkills();
+        ctrl.itemsPerPage = 1;
+        ctrl.skillPageNumber = 1;
+        var paginateSkillSpy = spyOn(ctrl, 'goToPageNumber');
+        ctrl.more = false;
+        ctrl.fetchSkills();
+        expect(paginateSkillSpy).toHaveBeenCalled();
+      });
+
+
+    it('should paginate forward without fetching if skills are present',
+      function() {
+        expect(ctrl.activeTab).toEqual('skills');
+        ctrl.more = false;
+        spyOn(ctrl, 'isNextSkillPagePresent').and.returnValue(true);
+        var paginateSkillSpy = spyOn(ctrl, 'goToPageNumber');
+        ctrl.navigateSkillPage('next_page');
+        expect(paginateSkillSpy).toHaveBeenCalled();
```

Comment:
Instead of just checking if this was called, can we make sure that the page number increased or something like that?

Reply:
Yeah, that looks better. Done